app/queries/create_quiz_submission_mutator.rb

```diff
@@ -10,21 +11,24 @@ class CreateQuizSubmissionMutator < ApplicationQuery
   validate :ensure_submittability
 
   def create_submission
-    TimelineEvent.transaction do
-      submission = target.timeline_events.create!(
+    submission =
+      TimelineEvent.transaction do
+      target.timeline_events.create!(
         checklist: result[:checklist],
         quiz_score: result[:score],
         passed_at: Time.zone.now
-      )
+      ).tap do |submission|
+        students.map do |student|
+          student.timeline_event_owners.create!(timeline_event: submission, latest: true)
+        end
 
-      students.map do |student|
-        student.timeline_event_owners.create!(timeline_event: submission, latest: true)
+        TimelineEvents::AfterMarkingAsCompleteJob.perform_later(submission)
       end
+    end
 
-      TimelineEvents::AfterMarkingAsCompleteJob.perform_later(submission)
+    publish(course, :submission_verified, current_user, submission)
```

Comment:
I would reuse the `:submission_automatically_verified` event here because there is very little difference between the event that happens here in `CreateQuizSubmissionMutator` vs the one in `AutoVerifySubmissionMutator`.

When using either mutator, the target is always marked as complete; the difference between the mutations is that this one records a response to the target's quiz, whereas the other one is used for targets that are simply marked as complete, or where the student is redirected to a link, before it's automatically marked as complete.

Reply:
> I would reuse the :submission_automatically_verified event here because there is very little difference ...

Ok, I'm still learning this domain. Changing it now.

> When using either mutator, the target is always marked as complete;...

Are there any places when target is marked as complete? Does `:submission_graded` also means target is completed? Does it depends on grade ? Is target completed if grade is < passing_grade ?

Reply:
> Are there any places when target is marked as complete?

I can't think of any other place where a target is completed through a `:submission_automatically_verified` event. 🤔 

> Does :submission_graded also means target is completed? Does it depends on grade ? Is target completed if grade is < passing_grade ?

Yes, it does depend on the grade, in exactly the same way that you described.

However, you should be aware of #347, where I proposed the removal of the concept of a failing grade. We've noticed that no one actually uses multiple levels of failure, so it'll be replaced with a simpler system where coaches can reject a submission outright without bother to _grade_ it.

Reply:
Thanks! I've subscribed for notifications in #347